modules/candidate_parameters/php/candidatequeryengine.class.inc

```diff
@@ -207,6 +290,39 @@ class CandidateQueryEngine extends \LORIS\Data\Query\SQLQueryEngine
                 . ' ON (c.RegistrationCenterID=rsite.CenterID)'
             );
             return 'rsite.Name';
+        case 'DateRegistered':
+            $this->addTable(
+                "LEFT JOIN session s ON (s.CandidateID=c.CandID AND s.Active='Y')"
```

Comment:
Wrong join condition.. CandidateID points to candidate.ID, not candidate.CandID. (same for all the other ones)